app/assets/javascripts/discourse/app/components/post/menu.gjs

```diff
@@ -0,0 +1,675 @@
+    return this.registeredButtons.get(POST_MENU_REPLIES_BUTTON_KEY);
+  }
+
+  get showMoreButton() {
+    return this.registeredButtons.get(POST_MENU_SHOW_MORE_BUTTON_KEY);
+  }
+
+  get remainingLikedUsers() {
+    return this.totalLikedUsers - this.likedUsers.length;
+  }
+
+  get remainingReaders() {
+    return this.totalReaders - this.readers.length;
+  }
+
+  @action
+  async like({ onBeforeToggle } = {}) {
+    if (!this.currentUser) {
+      this.keyValueStore &&
+        this.keyValueStore.set({
+          key: "likedPostId",
+          value: this.args.post.id,
+        });
+
+      this.args.showLogin();
+      return;
+    }
+
+    if (this.capabilities.userHasBeenActive && this.capabilities.canVibrate) {
+      navigator.vibrate(VIBRATE_DURATION);
+    }
+
+    if (this.args.post.liked) {
+      await this.#toggleLike();
+      return;
+    }
+
+    onBeforeToggle?.(this.args.post.liked);
+
+    return new Promise((resolve) => {
+      discourseLater(async () => {
+        await this.#toggleLike();
+        resolve();
+      }, 400);
+    });
+  }
+
+  @action
+  openAdminMenu(_, event) {
+    this.menu.show(event.target, {
+      identifier: "admin-post-menu",
+      component: AdminPostMenu,
+      modalForMobile: true,
+      autofocus: true,
+      data: {
+        post: this.args.post,
+        changeNotice: this.args.changeNotice,
+        changePostOwner: this.args.changePostOwner,
+        grantBadge: this.args.grantBadge,
+        lockPost: this.args.lockPost,
+        permanentlyDeletePost: this.args.permanentlyDeletePost,
+        rebakePost: this.args.rebakePost,
+        showPagePublish: this.args.showPagePublish,
+        togglePostType: this.args.togglePostType,
+        toggleWiki: this.args.toggleWiki,
+        unhidePost: this.args.unhidePost,
+        unlockPost: this.args.unlockPost,
+      },
+    });
+  }
+
+  @action
+  refreshReaders() {
+    if (this.readers.length) {
+      return this.#fetchWhoRead();
+    }
+  }
+
+  @action
+  showDeleteTopicModal() {
+    this.modal.show(DeleteTopicDisallowedModal);
+  }
+
+  @action
+  async showMoreActions() {
+    this.collapsed = false;
+
+    const fetchData = [
+      !this.showWhoLiked && this.#fetchWhoLiked(),
+      !this.showWhoRead && this.args.showReadIndicator && this.#fetchWhoRead(),
+    ].filter(Boolean);
+
+    await Promise.all(fetchData);
+  }
+
+  @action
+  toggleWhoLiked() {
+    if (this.showWhoLiked) {
+      this.showWhoLiked = false;
+      return;
+    }
+
+    this.#fetchWhoLiked();
+  }
+
+  @action
+  toggleWhoRead() {
+    if (this.showWhoRead) {
+      this.showWhoRead = false;
+      return;
+    }
+
+    this.#fetchWhoRead();
+  }
+
+  @action
+  async #toggleLike() {
+    await this.args.toggleLike();
+    if (!this.collapsed) {
+      await this.#fetchWhoLiked();
+    }
+  }
+
+  get #configuredItems() {
+    return this.siteSettings.post_menu.split("|").filter(Boolean);
+  }
+
+  @computed("args.post.bookmarked")
```

Comment:
We should avoid using legacy `@computed` in new code.

If you do `this.args.post.get("bookmarked")`, then it should autotrack properly

(or even better, add `@tracked bookmarked` to the post model)

Reply:
Done